proto/delivery/blender.proto

```diff
@@ -219,36 +336,61 @@ message BlenderConfig {
 // Next ID = 2.
 message QualityScoreConfig {
+    // A list of quality score terms. These are always summed.
   repeated QualityScoreTerm weighted_sum_term = 1;
 }
 
-// Next ID = 14.
+// Next ID = 2.
+message QualityScoreTerms {
+  // A list of quality score terms. These are always multiplied.
+  repeated QualityScoreTerm quality_score_terms = 1;
+}
+
+// Next ID = 15.
 message QualityScoreTerm {
   // How to fetch the base values of this term vector.
+  //
+  // If `fetch_method` is null, `attribute_name`, or `hashed_attribute` and the
+  // value cannot be retrieved from an insertion's sparse map, then this term
+  // will be set to `0.0`. Note: this will cause the product that this term is
+  // a part of to evaluate to 0.0.
   oneof fetch_method {
     // A named vector provided from elsewhere.
     string attribute_name = 1;
-    // Randomly generated values from a normal distribution.
-    NormalDistribution random_normal = 2;
-    // A constant value of ones. Set to any constant with offset and/or weight.
-    // Set to "true" to indicate that this option is set by convention.
-    bool ones = 3;
-    // Recursively evaluate terms and multiply their evaluations.
-    QualityScoreTerm multiply_term = 4;
+    // Recursively evaluate terms and multiply their evaluations as a product.
+    QualityScoreTerms product = 3;
   }
-  reserved 6 to 9; // Reserving fields for future fetch_method options.
+  reserved 4 to 9; // Reserving fields for future fetch_method options.
   // Maximum limit of underlying value (before weight and offset).
-  optional double fetch_high = 10;
+  optional float fetch_high = 10;
   // Minimum limit of underlying value (before weight and offset).
-  optional double fetch_low = 11;
-  // Multiply by this value. default =1 (no multiply).
-  double weight = 12;
-  // Add by this value. default = 0 (no addition)
-  double offset = 13;
+  optional float fetch_low = 11;
+  // Multiply by this value.
+  // Default: 1.0 (no multiply).
+  optional float weight = 12;
+  // Add a constant to this term (after applying the weight).
+  // Default: 0
+  float offset = 13;
+  // The value that this term evaluates to if the provided eval method evaluates to false.
+  // Default: ignored
+  optional TermConditionalEvaluation term_conditional_evaluation = 14;
 }
 
-// Next ID = 3.
-message NormalDistribution {
-  double mean = 1;
-  double variance = 2;
+// If specified, then the condition must evaluate to True. Otherwise, this
+// term evaluates to "value"
+// Use to define terms specific to different item types with different
+// Features. For example: video vs images.
+// Next Id 12;
+message TermConditionalEvaluation {
+  float value_if_false = 1;
+  string attribute_name = 2;
+  uint64 hashed_attribute = 3;
+  oneof eval_method {
+    Boolean boolean = 4;
+    Equal equal = 5;
+    GreaterThan greater_than = 6;
+    LessThan less_than = 7;
+    Interval interval = 8;
+    SparseIdMatch sparse_id_match = 9;
```

Comment:
This name is unfortunate. I have opened https://github.com/promotedai/schema-internal/pull/172 to deprecate it.